server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java

```diff
@@ -515,4 +520,89 @@ public void testDropPrimaryDuringReplication() throws Exception {
             verifyStoreContent();
         }
     }
+
+    public void testWaitUntilRefresh() throws Exception {
+        final String primaryNode = internalCluster().startNode();
+        createIndex(INDEX_NAME);
+        ensureYellowAndNoInitializingShards(INDEX_NAME);
+        final String replicaNode = internalCluster().startNode();
+        ensureGreen(INDEX_NAME);
+        final int initialDocCount = scaledRandomIntBetween(600, 700);
+        final List<ActionFuture<IndexResponse>> pendingIndexResponses = new ArrayList<>();
+        IndexShard primaryShard = getIndexShard(primaryNode, INDEX_NAME);
+        IndexShard replicaShard = getIndexShard(replicaNode, INDEX_NAME);
+
+        for (int i = 0; i < initialDocCount; i++) {
+            pendingIndexResponses.add(
+                client().prepareIndex(INDEX_NAME)
+                    .setId(Integer.toString(i))
+                    .setRefreshPolicy(WriteRequest.RefreshPolicy.WAIT_UNTIL)
+                    .setSource("field", "value" + i)
+                    .execute()
+            );
+        }
+        assertBusy(
+            () -> {
+                assertTrue(pendingIndexResponses.stream().allMatch(response -> response.actionGet().status().equals(RestStatus.CREATED)));
+            },
+            1,
+            TimeUnit.MINUTES
+        );
+
+        assertEquals(primaryShard.getLatestReplicationCheckpoint().getSeqNo(), replicaShard.getLatestReplicationCheckpoint().getSeqNo());
```

Comment:
Do we need to assert this with every wait_until request to ensure when request is Acked, the replica does catch up with latest seq no on primary ?

Reply:
We can do that. Do you thinksome assertion like this would be godd:
```
assertBusy(
            () -> {
                for(ActionFuture<IndexResponse> response:pendingIndexResponses){
                    if(response.actionGet().status().equals(RestStatus.CREATED)){
                        assertEquals(primaryShard.getLatestReplicationCheckpoint().getSeqNo(), replicaShard.getLatestReplicationCheckpoint().getSeqNo());
                    }
                }
                assertTrue(pendingIndexResponses.stream().allMatch(response -> response.actionGet().status().equals(RestStatus.CREATED)));
            },
            1,
            TimeUnit.MINUTES
        );
```

Reply:
I think it will be tricky with async operations here. There is one option of applying this assertion if we perform single indexing operation at a time, not sure if this is what we want from this test. Please feel free to ignore my previous comment here.

Reply:
Yes it will be tricky if we want to assert after every request. Th purpose of this test is to make sure that we wait until all requests are finished and replica has all indexed docs ready to be searched